bundle/edu.gemini.phase2.skeleton.servlet/src/main/scala/edu/gemini/phase2/skeleton/factory/SpProgramFactory.scala

```diff
@@ -215,6 +220,9 @@ object SpProgramFactory {
   def awardedHours(proposal: Proposal): Option[Double] =
     itacAcceptance(proposal) map { a => a.award.toHours.value }
 
+  private def hoursFromObservations(proposal: Proposal, sf: Observation => Option[TimeAmount]): TimeAmount =
+    proposal.observations.foldLeft(TimeAmount.empty)(_ |+| sf(_).getOrElse(TimeAmount.empty))
```

Comment:
There may be a nicer way to do this. I was first doing it with `map` and `flatten` to get rid of `None` in the `List[Option[TimeAmount]]`, but the compiler did not like `flatMap`.